src/Microsoft.AspNetCore.Razor.Language/Legacy/Span.cs

```diff
@@ -32,7 +33,32 @@ public Span(SpanBuilder builder)
 
         public override bool IsBlock => false;
 
-        public override int Length => Content.Length;
+        public override int Length
+        {
+            get
+            {
+                if (_length != -1)
+                {
+                    return _length;
+                }
+
+                var length = 0;
+                if (_content == null)
+                {
+                    for (var i = 0; i < Symbols.Count; i++)
+                    {
+                        length += Symbols[i].Content.Length;
+                    }
+                }
+                else
+                {
+                    length = _content.Length;
+                }
+
+                _length = length;
+                return length;
```

Comment:
This is kinda sloppy considering what this actually does.

This kinds of code is more more cleaner if written like:
```
if (_length == null)
{
   _length = ....
}

return _length;
```

If necessary turn your loop into a method.

Reply:
Also that will make it much more inlinable. Consider also sealing `Span`.